modules/collector-source/pkg/collector/collectordriver.go

```diff
@@ -0,0 +1,66 @@
+package collector
+
+import (
+	"sync/atomic"
+	"time"
+
+	"github.com/opencost/opencost/core/pkg/log"
+)
+
+type Config struct {
+	ScrapeInterval time.Duration
+}
+type CollectorDriver struct {
+	config     Config
+	isRunning  atomic.Bool
+	isStopping atomic.Bool
+	stop       chan struct{}
+	collector  MetricsCollector
+}
+
+func NewCollectorDriver(config Config) *CollectorDriver {
+	return &CollectorDriver{
+		collector: NewOpenCostMetricCollector(),
+	}
+}
+
+func (cd *CollectorDriver) Start() {
+	wasRunning := cd.isRunning.Swap(true)
+	if wasRunning {
+		log.Info("collector already running")
+		return
+	}
+	func() {
+		for {
+			select {
+			case <-cd.stop:
+				cd.isRunning.Store(false)
+				return
+			default:
+
+			}
+			time.Sleep(cd.config.ScrapeInterval)
+		}
```

Comment:
This blocks? You can drop the whole dual atomic bool and stop channel design for an `AtomicRunState` -- Example here: https://github.com/opencost/opencost/blob/cf4d510ad4dd1c27e37d7ce98a9b2c4296e2a402/core/pkg/exporter/controller.go#L63-L131

Reply:
That is definitely a better option, I never actually ended up plugging this in. I think the real question in my mind is do we actually even need start and stop for something like this.